### DISCUSSION TITLE
Actor stops without warning

### DISCUSSION THREAD

**Discussion by tshowan** [statelyai/xstate#4825] [Q&A]
For seemingly no reason my state machine is stopping, and sending further events gives me the error: 
```
Event "click on item" was sent to stopped actor "x:0 (x:0)". This actor has already reached its final state, and will not transition.
```

I have already discovered that this happens if there's an error in an Action. But it also seems to happen at random, presumably because something is going wrong somewhere else in the process. (I have established it's not my Action by simply doing `console.log("finished action");` as the final statement.)

Working without error messages is making this development process needlessly frustrating. And when the error happens from somewhere outside my sphere of control, without error messages I have no way to even guess what's going wrong.

Why are there no error messages, are you swallowing them somehow, and is there a setting I can use to allow them through?

**Comment by Andarist** [comment]
Please always try to share a repro case in a runnable form - either by providing a git repository to clone or a codesandbox.

**Reply by tshowan** [reply]
In this case I'm not asking you to debug my error. I want to know why I can't see error messages in order to debug it myself.

**Reply by Andarist** [reply]
It's a good question but I don't know the answer unless you show me the repro case of the problem. Most errors should be rethrown in the clear stack and you should be able to observe them in the console as normal. You can also try to pause on caught exceptions in the debugger to learn more about what is thrown and where.

**Reply by tshowan** [reply]
So, to clarify, right now I have:
```js
updateEntryData:  ({ context, event }) => {
            console.log("updateEntryData STARTED");
            let a = 5;
            a.b.c = 5;
            console.log("updateEntryData FINISHED");
}
```
When I trigger that action I expect to see `updateEntryData STARTED` appear in the console, followed by an error.
Instead I see `updateEntryData STARTED` only, and next time I send my event I get the message that the actor stopped.

Can you confirm that what I'm seeing is not expected behaviour?

**Reply by Andarist** [reply]
From what I can see it errors quite alright here: https://stackblitz.com/edit/typescript-nmabbh?file=index.ts
